src/utils/api.error.js

```diff
@@ -0,0 +1,44 @@
+class ApiError extends Error {
+  status;
+  errors;
+
+  constructor({ message, status, errors = {} }) {
+    super(message);
+    this.status = status;
+    this.errors = errors;
+  }
+
+  static badRequest(message, errors = {}) {
+    return new ApiError({
+      message,
+      errors,
+      status: 400,
+    });
+  }
+
+  static unauthorized(errors = {}) {
+    return new ApiError({
+      message: 'unauthorized user',
+      errors,
+      status: 401,
+    });
+  }
+
+  static forbidden(message, errors = {}) {
+    return new ApiError({
+      message: 'resource is forbidden',
```

Comment:
Consider allowing a custom message to be passed in for the `forbidden` method to provide more context and maintain consistency with other methods.